libwallet/assets/dcr/dex_asset.go

```diff
@@ -0,0 +1,511 @@
+// Part of the Wallet interface.
+func (da *DEXAsset) Connect(_ context.Context) error {
+	return nil
+}
+
+// Disconnect shuts down access to the wallet.
+// Part of the Wallet interface.
+func (da *DEXAsset) Disconnect() {}
+
+// SpvMode returns true if the wallet is connected to the Decred
+// network via SPV peers.
+// Part of the Wallet interface.
+func (da *DEXAsset) SpvMode() bool {
+	return true
+}
+
+// NotifyOnTipChange registers a callback function that the should be
+// invoked when the wallet sees new mainchain blocks. The return value
+// indicates if this notification can be provided. Where this tip change
+// notification is unimplemented, monitorBlocks should be used to track
+// tip changes. TODO: Use tipFeed if it's exported.
+// Part of the Wallet interface.
+func (da *DEXAsset) NotifyOnTipChange(ctx context.Context, cb dexdcr.TipChangeCallback) bool {
+	const dexAssetOnTipCallbackIdentifier = "dex-asset-on-tip-change"
+	listener := &wallet.TxAndBlockNotificationListener{
+		OnBlockAttached: func(walletID int, blockHeight int32) {
+			cb(ctx, nil, int64(blockHeight), nil)
+		},
+	}
+
+	da.asset.RemoveTxAndBlockNotificationListener(dexAssetOnTipCallbackIdentifier) // Clear previous listener if any.
+	err := da.asset.AddTxAndBlockNotificationListener(listener, dexAssetOnTipCallbackIdentifier)
+	return err != nil // TODO
+}
+
+// AddressInfo returns information for the provided address. It is an error
+// if the address is not owned by the wallet.
+// Part of the Wallet interface.
+func (da *DEXAsset) AddressInfo(ctx context.Context, address string) (*dexdcr.AddressInfo, error) {
+	addr, err := stdaddr.DecodeAddress(address, da.asset.chainParams)
+	if err != nil {
+		return nil, err
+	}
+	ka, err := da.asset.Internal().DCR.KnownAddress(ctx, addr)
+	if err != nil {
+		return nil, err
+	}
+
+	if ka, ok := ka.(dcrwallet.BIP0044Address); ok {
+		_, branch, _ := ka.Path()
+		return &dexdcr.AddressInfo{Account: ka.AccountName(), Branch: branch}, nil
+	}
+	return nil, fmt.Errorf("unsupported address type %T", ka)
+}
+
+// AccountOwnsAddress checks if the provided address belongs to the
+// specified account.
+// Part of the Wallet interface.
+func (da *DEXAsset) AccountOwnsAddress(ctx context.Context, addr stdaddr.Address, acctName string) (bool, error) {
+	a, err := da.asset.Internal().DCR.KnownAddress(ctx, addr)
+	if err != nil {
+		return false, utils.TranslateError(err)
+	}
+
+	if a.AccountName() != acctName {
+		return false, nil
+	}
+
+	if kind := a.AccountKind(); kind != dcrwallet.AccountKindBIP0044 && kind != dcrwallet.AccountKindImported {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+// AccountBalance returns the balance breakdown for the specified account.
+// Part of the Wallet interface.
+func (da *DEXAsset) AccountBalance(_ context.Context, confirms int32, _ string) (*walletjson.GetAccountBalanceResult, error) {
+	bal, err := da.asset.GetAccountBalance(int32(da.dexAccountNumber), &confirms)
```

Comment:
I see why `asset.GetAccountBalance` was modified to accept confs now. Please use the internal wallet method for getting account balance. And use the provided ctx as well.